samples/WebsocketConnect/src/main/java/websocketconnect/WebsocketConnect.java

```diff
@@ -0,0 +1,135 @@
+        }
+    }
+
+    public static void main(String[] args) {
+
+        cmdUtils = new CommandLineUtils();
+        cmdUtils.registerProgramName("WebsocketConnect");
+        cmdUtils.addCommonMQTTCommands();
+        cmdUtils.addCommonProxyCommands();
+        cmdUtils.registerCommand("signing_region", "<str>", "AWS IoT service region.");
+        cmdUtils.registerCommand("client_id", "<int>", "Client id to use (optional, default='test-*').");
+        cmdUtils.registerCommand("port", "<int>", "Port to connect to on the endpoint (optional, default='8883').");
+        cmdUtils.sendArguments(args);
+
+        String endpoint = cmdUtils.getCommandRequired("endpoint", "");
+        String clientId = cmdUtils.getCommandOrDefault("client_id", "test-" + UUID.randomUUID().toString());
+        int port = Integer.parseInt(cmdUtils.getCommandOrDefault("port", String.valueOf(8883)));
+        String caPath = cmdUtils.getCommandOrDefault("ca_file", "");
+        String signingRegion = cmdUtils.getCommandRequired("signing_region", "");
+        String proxyHost = cmdUtils.getCommandOrDefault("proxy_host", "");
+        int proxyPort = Integer.parseInt(cmdUtils.getCommandOrDefault("proxy_port", "8080"));
+
+        MqttClientConnectionEvents callbacks = new MqttClientConnectionEvents() {
+            @Override
+            public void onConnectionInterrupted(int errorCode) {
+                if (errorCode != 0) {
+                    System.out.println("Connection interrupted: " + errorCode + ": " + CRT.awsErrorString(errorCode));
+                }
+            }
+
+            @Override
+            public void onConnectionResumed(boolean sessionPresent) {
+                System.out.println("Connection resumed: " + (sessionPresent ? "existing session" : "clean session"));
+            }
+        };
+
+        try (
+            AwsIotMqttConnectionBuilder builder = AwsIotMqttConnectionBuilder.newMtlsBuilderFromPath(null, null)) {
+
+            if (caPath != null) {
+                builder.withCertificateAuthorityFromPath(null, caPath);
+            }
+
+            builder.withConnectionEventCallbacks(callbacks)
+                    .withClientId(clientId)
+                    .withEndpoint(endpoint)
+                    .withPort((short)port)
+                    .withCleanSession(true)
+                    .withProtocolOperationTimeoutMs(60000);
+
+            HttpProxyOptions proxyOptions = null;
+            if (proxyHost != null && proxyHost != "" && proxyPort > 0) {
+                proxyOptions = new HttpProxyOptions();
+                proxyOptions.setHost(proxyHost);
+                proxyOptions.setPort(proxyPort);
+                builder.withHttpProxyOptions(proxyOptions);
+            }
+
+            builder.withWebsockets(true);
+            builder.withWebsocketSigningRegion(signingRegion);
+
+            try(MqttClientConnection connection = builder.build()) {
```

Comment:
Why doesn't this use buildWebsocketConnection?

Reply:
I think this was due to a misunderstanding on my part. I thought I needed to have the connect samples show the full method of connection without the utility functions. I'll adjust the samples here (and in C++ and Python) to use the simplified builders in the CmdUtils class. 👍